src/lib/ModelConfig.cc

```diff
@@ -135,8 +135,39 @@ void ModelConfig::validate() {
   }
   subFields.clear();
 
-  // L1-Cache
-  root = "L1-Cache";
+  // Instruction Memory
+  root = "Instruction-Memory";
+  subFields = {"Interface-Type"};
+  nodeChecker<std::string>(
+      configFile_[root][subFields[0]], root + " " + subFields[0],
```

Comment:
Not quite sure what you mean.

There are two Interface-Type fields in each config file, hence, we need a prefix to tell the user which one it's referencing. Hopefully that answers your query.

Reply:
If you look towards another nodeChecker<string> call, i.e. simulation mode, the second argument is `subFields[0]` rather than `root + " " + subFields[0]`. Why is it different for the interface types?

Reply:
> There are two Interface-Type fields in each config file, hence, we need a prefix to tell the user which one it's referencing. Hopefully that answers your query.

We need the extra root prefix to distinguish the two config params. Taking your example of "simulation mode", there only exists one "simulation mode" param and thus doesn't need the extra prefix.

Reply:
Makes sense, thanks for clarification